fhirflat/resources/condition.py

```diff
@@ -0,0 +1,119 @@
+from fhir.resources.condition import Condition
+import orjson
+from itertools import groupby
+
+import pandas as pd
+
+from ..fhir2flat import fhir2flat
+from ..flat2fhir import expand_concepts
+
+# TODO: Update references to disallow "display" (could contain names)
+
+
+class Condition(Condition):
+    # attributes to exclude from the flat representation
+    flat_exclusions: set[str] = (
+        "meta",
+        "implicitRules",
+        "language",
+        "text",
+        "contained",
+        "modifierExtension",
+        "identifier",
+        "verificationStatus",
+        "evidence",
+    )
+
+    # required attributes that are not present in the FHIRflat representation
+    flat_defaults: list[str] = ["clinicalStatus"]
+
+    @property
+    def attr_lists(self):
+        """Attributes which take a list of FHIR types."""
+        return [
+            p.alias
+            for p in self.element_properties()
+            if "typing.List" in str(p.outer_type_)
+        ]
+
+    @classmethod
+    def from_flat(cls, file):
+        """ "
+        Takes a FHIRflat pandas dataframe and populates the resource with the data.
+
+        file: parquet file
+            FHIRflat file containing patient data
+        """
+
+        def cleanup(cls, data):
+            """
+            Load data into a dictionary-like structure, then
+            apply resource-specific changes and unpack flattened data
+            like codeableConcepts back into structured data.
+            """
+            data = orjson.loads(data)
+
+            data["encounter"] = {"reference": data["encounter"]}
+            data["subject"] = {"reference": data["subject"]}
+
+            # add default clinicalStatus back in
+            data["clinicalStatus"] = {
+                "coding": [
+                    {
+                        "system": "http://terminology.hl7.org/CodeSystem/condition-clinical",
+                        "code": "unknown",
+                    }
+                ]
+            }
+
+            # find and group keys belonging to the same concept
+            grouped_keys = [k for k in data.keys() if "." in k]
+            grouped_keys.sort()
+            groups = [
+                {k: [gs for gs in g]}
+                for k, g in groupby(grouped_keys, lambda x: x.split(".")[0])
+            ]
+            data = expand_concepts(data, groups)
+
+            # create lists for properties which are lists of FHIR types
+            # FIXUP: This is not elegant, or fast
```

Comment:
what is the speed hit here?

Reply:
Intuition rather than anything I've checked specifically